### DISCUSSION TITLE
Using `ignoredActionPaths` and `ignoredPaths` with RTK Query

### DISCUSSION THREAD

**Discussion by joekrall** [reduxjs/redux-toolkit#1667] [Q&A] [answered]
I am attempting to add certain action paths to `ignoredPaths` and `ignoredActionPaths` while using RTK Query. I am able to successfully ignore those paths, but in doing so, I am getting the following errors, apparently from RTK Query:

```
A non-serializable value was detected in an action, in the path: `meta.baseQueryMeta.request`. Value: 
Request { method: "POST", url: "http://localhost:3000/api/users/login", headers: Headers, destination: "", referrer: "about:client", referrerPolicy: "", mode: "cors", credentials: "same-origin", cache: "default", redirect: "follow" }
 
Take a look at the logic that dispatched this action:  
Object { type: "api/executeMutation/fulfilled", payload: {…}, meta: {…} }
 
(See https://redux.js.org/faq/actions#why-should-type-be-a-string…t-least-serializable-why-should-my-action-types-be-constants) 
(To allow non-serializable values see: https://redux-toolkit.js.org/usage/usage-guide#working-with-non-serializable-data) 

A non-serializable value was detected in an action, in the path: `meta.baseQueryMeta.request`. Value: 
Request { method: "GET", url: "http://localhost:3000/api/campus", headers: Headers, destination: "", referrer: "about:client", referrerPolicy: "", mode: "cors", credentials: "same-origin", cache: "default", redirect: "follow" }
 
Take a look at the logic that dispatched this action:  
Object { type: "api/executeQuery/fulfilled", payload: (155) […], meta: {…} }
 
(See https://redux.js.org/faq/actions#why-should-type-be-a-string…t-least-serializable-why-should-my-action-types-be-constants) 
(To allow non-serializable values see: https://redux-toolkit.js.org/usage/usage-guide#working-with-non-serializable-data)
```

Here is my current store.

```
export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({ 
      serializableCheck: {
      // Ignore all the action types redux-persist dispatches:
      // https://redux-toolkit.js.org/usage/usage-guide#use-with-redux-persist
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        ignoredActionPaths: ['meta.arg', 'payload.centerPnt'],
        ignoredPaths: ['campusMapReducer.circle.centerPnt'],
      },
      // This enables caching, invalidation, polling, and other features of RTK Query
    }).concat(api.middleware),
    preloadedState,
  });
};
```
Do I now need to add `meta.baseQueryMeta.request` and `meta.baseQueryMeta.response`, etc., explicitly to `ignoredActionPaths`?

**Comment by markerikson** [comment, marked as answer]
Yeah, the default values are: `ignoredActionPaths = ['meta.arg', 'meta.baseQueryMeta'],`.  If you're providing your own array, you probably should make sure those two values are in there also.

**Reply by joekrall** [reply]
Thank you! Currently, the [section on Serializability Middleware ](https://redux-toolkit.js.org/api/serializabilityMiddleware#options)in the docs only lists `[meta.arg]`. Could I contribute an update to the documentation?

**Reply by markerikson** [reply]
Please do! The `meta.baseQueryMeta` was only added to the code in 1.6 with the release of RTK Query, so obviously we missed updating that bit of the docs.

**Reply by joekrall** [reply]
I just opened a new issue (#1669)!

**Reply by alexandernst** [reply]
Maybe `ignoredActionPaths` should be an array or a function? If function is provided, execute it passing the current array of ignored action paths and get the newly returned array.

**Reply by EskiMojo14** [reply]
i'd be open to a `ignoredActionPaths: defaultPaths => defaultPaths.concat("extra")` callback - for consistency i wonder if it's worth supporting the same for `ignoredActions` and `ignoredPaths` even though those currently don't have any defaults
which then also raises the question of doing the same for the immutability middleware

**Reply by alexandernst** [reply]
I'd say yes, if nothing else just for consistency.
